ITKImageProcessingFilters/itkInPlaceImageToDream3DDataFilter.hxx

```diff
@@ -0,0 +1,240 @@
+  IGeometry::Pointer geom = dataContainer->getGeometry();
+  if (geom)
+  {
+    imageGeom = std::dynamic_pointer_cast<ImageGeom>(geom);
+    if (!imageGeom)
+    {
+      itkExceptionMacro("Data container contains a geometry that is not ImageGeometry");
+    }
+    imageGeom->getOrigin(torigin[0], torigin[1], torigin[2]);
+    imageGeom->getResolution(tspacing[0], tspacing[1], tspacing[2]);
+    imageGeom->getDimensions(tDims[0], tDims[1], tDims[2]);
+    for( size_t i=0; i < 3; i++ )
+    {
+      if (abs(torigin[i] - origin[i]) > tol
+        || abs(tspacing[i] - spacing[i]) > tol
+        || tDims[i] != size[i]
+        )
+      {
+        itkExceptionMacro("Data container contains a geometry that do not match image properties");
+      }
+    }
+  }
+  else
+  {
+    imageGeom = ImageGeom::CreateGeometry(SIMPL::Geometry::ImageGeometry);
+    if (!imageGeom)
+    {
+      itkExceptionMacro("Could not create image geometry");
+    }
+    for (size_t i = 0; i < VDimension; i++)
+    {
+      torigin[i] = origin[i];
+      tspacing[i] = spacing[i];
+      tDims[i] = size[i];
+    }
+    imageGeom->setOrigin(torigin[0], torigin[1], torigin[2]);
+    imageGeom->setResolution(tspacing[0], tspacing[1], tspacing[2]);
+    imageGeom->setDimensions(tDims[0], tDims[1], tDims[2]);
+    dataContainer->setGeometry(imageGeom);
+  }
+}
+
+template<typename PixelType, unsigned int VDimension>
+void
+InPlaceImageToDream3DDataFilter<PixelType, VDimension>
+::GenerateData()
+{
+  DecoratorType *outputPtr = this->GetOutput();
+  DataContainer::Pointer dataContainer = outputPtr->Get();
+  ImagePointer inputPtr = dynamic_cast<ImageType*>(this->GetInput(0));
+  // Create data array
+  QVector<size_t> cDims(1, 0);
+  cDims[0] = 1;
+  IGeometry::Pointer geom = dataContainer->getGeometry();
+  ImageGeom::Pointer imageGeom = std::dynamic_pointer_cast<ImageGeom>(geom);
+  size_t dims[3];
+  imageGeom->getDimensions(dims[0], dims[1], dims[2]);
+  QVector<size_t> tDims(3, 1);
+  qCopy(dims, dims + 3, tDims.begin());
+  AttributeMatrix::Pointer ma;
+  if (dataContainer->doesAttributeMatrixExist(m_MatrixArrayName.c_str()))
+  {
+    // Check that size matches
+    ma = dataContainer->getAttributeMatrix(m_MatrixArrayName.c_str());
+    size_t numofTuples = ma->getNumTuples();
+    QVector<size_t> matDims = ma->getTupleDimensions();
+    if (matDims != tDims)
+    {
+      itkExceptionMacro("Tuples dimension of existing matrix array do not match image size.");
+    }
+    if (ma->getType() != SIMPL::AttributeMatrixType::Cell)
+    {
+      itkExceptionMacro("Attribute matrix is not of type SIMPL::AttributeMatrixType::Cell.");
+    }
+  }
+  else
+  {
+    ma = dataContainer->createAndAddAttributeMatrix(tDims, m_MatrixArrayName.c_str(), SIMPL::AttributeMatrixType::Cell);
+  }
+  // Checks if doesAttributeArray exists
+  if (ma->doesAttributeArrayExist(m_DataArrayName.c_str()))
+  {
+    itkExceptionMacro("Existing data array with same name.");
+  }
```

Comment:
Could we add a boolean member variable to this class that determines whether this check is run?  During a DREAM.3D filter's preflight, we will often stick arrays into the data structure to show up in the user interface so that the user knows what arrays will be created/removed; in this special case, we'll want to create 'dummy' arrays during this step that get overwritten later on by the itkInPlaceImageToDream3DDataFilter class.  Having an option here to either do this check or not gives us the flexibility to operate either way.